src/rust_frontend/vf_mir_translator/rust_parser.ml

```diff
@@ -8,9 +9,10 @@ let is_expr_with_block = function
 | SwitchExpr (_, _, _, _) -> true
 | _ -> false
 
-let rec parse_type_path_rest x = function%parser
-  [ (_, Kwd "::"); (_, Ident xx); [%let x = parse_type_path_rest (x ^ "::" ^ xx)] ] -> x
-| [ ] -> x
+let rec parse_type_path_rest l x = function%parser
```

Comment:
Should `parse_type_path_rest` be renamed to `parse_path_rest` (or `parse_simple_path_rest`?

Reply:
Also, insert something into your parser_test.rs that exercises this?